src/ape/api/providers.py

```diff
@@ -250,13 +337,22 @@ class BlockAPI:
     @classmethod
     @abstractmethod
     def decode(cls, data: Dict) -> "BlockAPI":
-        ...
+        """
+        Decode data to a :class:`~ape.api.BlockAPI`.
+
+        Args:
+            data (dict): A dictionary of data to decode.
+
+        Returns:
+            :class:`~ape.api.BlockAPI`
+        """
 
 
 @abstractdataclass
 class ProviderAPI:
     """
-    A Provider must work with a particular Network in a particular Ecosystem
+    A Provider must work with a particular Network in a particular Ecosystem.
```

Comment:
```suggestion
    A Provider must work with a particular network in a particular Ecosystem.
```

network does not need capitalization